File: BackEnd_Learn/Java/chat/src/main/java/com/example/chat/Security/Jwt/Auth/AuthDetails.java
package com.example.chat.Security.Jwt.Auth;

import com.example.chat.Entity.User.User;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.ArrayList;
import java.util.Collection;

@Getter
@AllArgsConstructor
public class AuthDetails implements UserDetails {

    private final User user;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return new ArrayList<>();
    }

    @Override
    public String getPassword() {
        return user.getUserPw();
    }


    @Override
    public String getUsername() {
        return user.getUserId();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: BackEnd_Learn/kotlin/entity/src/main/kotlin/com/example/entity/security/jwt/auth/AuthDetails.kt
package com.example.entity.security.jwt.auth

import com.example.entity.entity.user.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.*
import kotlin.collections.ArrayList

class AuthDetails(val user: Optional<User>): UserDetails {


    override fun getAuthorities(): MutableCollection<out GrantedAuthority> {
        return ArrayList<GrantedAuthority>()
    }

    override fun isEnabled(): Boolean {
        return true
    }

    override fun getUsername(): String {
        return user
                .map { t: User -> t.name }.get()
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun getPassword(): String {
        return user
                .map { t: User -> t.password }.get()
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }
}